sqlite3/lib/src/ffi/constants.dart

```diff
@@ -1,44 +1,247 @@
+
+  /// Ok for sqlite3_open_v2()
+  static const int SQLITE_OPEN_NOMUTEX = 0x00008000;
+
+  /// Ok for sqlite3_open_v2()
+  static const int SQLITE_OPEN_FULLMUTEX = 0x00010000;
+
+  /// Ok for sqlite3_open_v2()
+  static const int SQLITE_OPEN_SHAREDCACHE = 0x00020000;
+
+  /// Ok for sqlite3_open_v2()
+  static const int SQLITE_OPEN_PRIVATECACHE = 0x00040000;
+
+  /// VFS only
+  static const int SQLITE_OPEN_WAL = 0x00080000;
+}
 
 // Prepare flags, https://www.sqlite.org/c3ref/c_prepare_normalize.html
-const SQLITE_PREPARE_PERSISTENT = 0x01;
-const SQLITE_PREPARE_NO_VTAB = 0x04;
+class SqlPrepareFlags {
+  ///The SQLITE_PREPARE_PERSISTENT flag is a hint to the query planner that the prepared statement will be retained for a long time and probably reused many times.
+  /// Without this flag, sqlite3_prepare_v3() and sqlite3_prepare16_v3() assume that the prepared statement will be used just once or at most a few times and then destroyed using sqlite3_finalize() relatively soon.
+  ///  The current implementation acts on this hint by avoiding the use of lookaside memory so as not to deplete the limited store of lookaside memory.
+  ///Future versions of SQLite may act on this hint differently.
+  static const int SQLITE_PREPARE_PERSISTENT = 0x01;
+
+  ///The SQLITE_PREPARE_NORMALIZE flag is a no-op. This flag used to be required for any prepared statement that wanted to use the sqlite3_normalized_sql() interface.
+  /// However, the sqlite3_normalized_sql() interface is now available to all prepared statements, regardless of whether or not they use this flag.
+  static const int SQLITE_PREPARE_NORMALIZE = 0x02;
+
+  ///the SQL compiler to return an error (error code SQLITE_ERROR) if the statement uses any virtual tables
+  static const int SQLITE_PREPARE_NO_VTAB = 0x04;
+}
+
+/// Datatypes, https://sqlite.org/c3ref/c_blob.html
+class SqlTypes {
+  static const int SQLITE_INTEGER = 1;
+  static const int SQLITE_FLOAT = 2;
+  static const int SQLITE_TEXT = 3;
+  static const int SQLITE_BLOB = 4;
+  static const int SQLITE_NULL = 5;
+}
+
+/// Text Encodings, https://www.sqlite.org/c3ref/c_any.html
+/// These constant define integer codes that represent the various text encodings supported by SQLite.
+class SqlTextEncodings {
+  ///IMP: R-37514-35566
+  static const int SQLITE_UTF8 = 1;
+
+  ///IMP: R-03371-37637
+  static const int SQLITE_UTF16LE = 2;
+
+  ///IMP: R-51971-34154
+  static const int SQLITE_UTF16BE = 3;
+
+  ///Use native byte order
+  static const int SQLITE_UTF16 = 4;
+
+  ///Deprecated
+  static const int SQLITE_ANY = 5;
 
-// Datatypes, https://sqlite.org/c3ref/c_blob.html
+  ///qlite3_create_collation only
+  static const int SQLITE_UTF16_ALIGNED = 8;
+}
 
-const SQLITE_INTEGER = 1;
-const SQLITE_FLOAT = 2;
-const SQLITE_TEXT = 3;
-const SQLITE_BLOB = 4;
-const SQLITE_NULL = 5;
+/// Special destructors, https://www.sqlite.org/c3ref/c_static.html
+class SqlSpecialDestructor {
```

Comment:
Nice call to introduce fields for these values! Let's just not expose them to users.

Reply:
You are right, I've changed the visibility in sqlite3.dart on the last commit